tests/std/tests/VSO_0000000_allocator_propagation/test.cpp

```diff
@@ -34,82 +34,106 @@ _CONSTEXPR20 void assert_is_permutation(const Container& cont, initializer_list<
     assert(is_permutation(cont.begin(), cont.end(), il.begin(), il.end()));
 }
 
+int64_t pointer_counter = 0;
+_CONSTEXPR20 void increase_counter() {
+    if (!_Is_constant_evaluated()) {
+        ++pointer_counter;
+    }
+}
+_CONSTEXPR20 void decrease_counter() {
+    if (!_Is_constant_evaluated()) {
+        --pointer_counter;
+        assert(pointer_counter >= 0);
+    }
+}
+
+constexpr uintptr_t living_word = 0x86427531;
+

```

Comment:
`living` and `living_word` (I need to find some better names for them) are used to detect that the pointer doesn't work in unconstructed state. A failed `assert` 100% suggests a bug; a successful `assert` suggests a high rate that the pointer really works in well-constructed state.

Reply:
As the `pointer_counter` is enough to detect both bugs (and it's not hard to locate those bugs with it), can I just remove `magic_number` and `dat`? This will make the code less noisy...